src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java

```diff
@@ -269,45 +270,44 @@ else if (status.getOrderStatus().contains("CANCELLED") && (StringUtils.isBlank(k
         }
     }
 
-    public static void processingSingleConfirmation(Response gbfResponse, List<KitRequest> kitRequests, ShippingConfirmation confirmation) throws Exception {
-        //update external shipper response at request
+    public static void processingSingleConfirmation(Response gbfResponse, ShippingConfirmation confirmation) throws Exception {
         logger.info("Got confirmation for " + confirmation.getOrderNumber());
         Node node = GBFRequestUtil.getXMLNode(gbfResponse.getXML(), XML_NODE_EXPRESSION.replace("%1", confirmation.getOrderNumber()));
-        List<String> dsmKitRequestIds = getDSMKitRequestId(confirmation.getOrderNumber());
-        if (dsmKitRequestIds != null && !dsmKitRequestIds.isEmpty()) {
-            for (String dsmKitRequestId : dsmKitRequestIds) {
-                KitRequestExternal.updateKitRequestResponse(getStringFromNode(node), dsmKitRequestId);
-            }
-        }
-        //update kit shipping information
-        if (confirmation.getItem() != null) {
+        String externalResponse = getStringFromNode(node);
+        if (confirmation.getItem() != null && StringUtils.isNotBlank(externalResponse)) {
             Item item = confirmation.getItem();
-            List<KitRequest> kitRequestsResult = getKitRequest(kitRequests, confirmation.getOrderNumber(), item.getItemNumber(), 1);
-            int counter = 0;
-            if (kitRequestsResult != null) {
-                for (KitRequest kitRequest : kitRequestsResult) {
-                    String kitLabel = item.getTubeSerial();
-                    if (counter > 0) {
-                        kitLabel += "_" + counter;
+            inTransaction((conn) -> {
```

Comment:
making now 1 connection for request and kit information update

Reply:
Is `item` always non-null?  Before, we had a check for this, so I'm wondering if something might NPE without it.

Reply:
Ah, I see the check up above--maybe move `Item item = confirmation.getItem();` before L277 and then change `confirmation.getItem() != null` to `item != null`?